api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -666,9 +683,169 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenFolderDoesNotContainFileWithGivenName() {
+		File folder = createFolderInUserDirectory("folder");
+		assertFalse(OpenmrsUtil.folderContains(folder,"file"));
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenGivenStringIsNull() {
+		File folder = createFolderInUserDirectory("folder");
+		assertFalse(OpenmrsUtil.folderContains(folder, null));
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenGivenFolderIsNull() {
+		assertFalse(OpenmrsUtil.folderContains(null,"file"));
+	}
+	
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnEmptyListWhenGivenDescriptorIsNullOrEmpty() {
+		List<Concept> list = OpenmrsUtil.conceptListHelper("");
+		assertTrue(list.size() == 0);
+
+		list = OpenmrsUtil.conceptListHelper(null);
+		assertTrue(list.size() == 0);
+	}
+
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnListOfConceptsWithGivenNameInDescriptor() {
+		List<Concept> helperList = OpenmrsUtil.conceptListHelper("name:YES|name:NO|name:DIED|set:25|set:26");
+		List<Concept> serviceList = new ArrayList<>();
+		List<Concept> aux;
+		Concept con;
+		//Adding the concepts in the list to compare with the first one
+		ConceptService cs = Context.getConceptService();
+		serviceList.add(cs.getConceptByName("YES"));
+		serviceList.add(cs.getConceptByName("NO"));
+		serviceList.add(cs.getConceptByName("DIED"));
+		
+		con = cs.getConcept(25);
+		aux = cs.getConceptsByConceptSet(con);
+		serviceList.addAll(aux);
+		
+		con = cs.getConcept(26);
+		aux = cs.getConceptsByConceptSet(con);
+		serviceList.addAll(aux);
+		
+		assertTrue("List should be same size to be equal",helperList.size() == serviceList.size());
+		//Removing common elements
+		Set<Concept> serviceSet = new HashSet<>(serviceList);
+		helperList.stream()
+				.forEach(concept -> {
+					serviceSet.remove(concept);
+				});
+		
+		assertTrue("The set should be empty" ,serviceSet.size() == 0);
+	}
+	
+	/**
+	 * @see OpenmrsUtil#getFileAsString(File)
+	 */
+	@Test
+	public void getFileAsString_shouldReturnTheStringContainedInFile() {
+		try {
+			String separator = FileSystems.getDefault().getSeparator();
+			File file = new File(System.getProperty("user.dir") + separator + "file");
+			file.createNewFile();
+			PrintWriter writer = new PrintWriter(file);
+			String output = "The first line" + "\n" + "The second line";
+			writer.println("The first line");
+			writer.print("The second line");
+			writer.close();
+			assertTrue(output.equals(OpenmrsUtil.getFileAsString(file)));
+			file.delete();
+			
+		}
+		catch (IOException e) {}
```

Comment:
What happens if the IOException is caught here?
Does that mean we shall have no test assertions?

Reply:
I can say something like catch (IOException e){Assert.fail();}

Reply:
Is this of help https://stackoverflow.com/questions/16596418/how-to-handle-exceptions-in-junit/16597164#16597164?